x-pack/plugins/security_solution/public/common/components/header_actions/actions.tsx

```diff
@@ -254,51 +251,28 @@ const ActionsComponent: React.FC<ActionProps> = ({
             />
           )}
         </>
-        {securitySolutionNotesEnabled && !expandableFlyoutDisabled && toggleShowNotes && (
+        {!isEventViewer && showNotes && (
```

Comment:
I think for this whole section below we need to support all these scenarios:

- nothing in `kibana.yml`
    - note icon only shows up in timeline
    - note icon should have dot if document has notes
    - should open the EuiFlyout

- `kibana.yml has only `securitySolutionNotesEnabled`
    - note icon shows everywhere (timeline, alerts page, all explore pages, alerts table in cases)
    - all note icons should have red dots if the document has notes
    - pin icon shows up only in timeline
    - should all open the expandable flyout

- `kibana.yml` has `securitySolutionNotesEnabled` and `expandableFlyoutDisabled` (this is the same as first scenario)
    - note icon only shows up in timeline
    - note icon should have dot if document has notes
    - should open the EuiFlyout

- `kibana.yml` has only `unifiedComponentsInTimelineEnabled` (this is the same as first and third scenario but with the new fancy timeline look)
    - note icon only shows up in timeline
    - note icon should have dot if document has notes
    - should open the EuiFlyout

- `kibana.yml` has `securitySolutionNotesEnabled` and `unifiedComponentsInTimelineEnabled` (this is the same as the secon scenario but with the new fancy timeline look)
    - note icon shows everywhere (timeline, alerts page, all explore pages, alerts table in cases)
    - all note icons should have red dots if the document has notes
    - pin icon shows up only in timeline
    - should all open the expandable flyout

I was so confused I had to list these scenario to make sure that I got them right (did I?) and also to help writing down the next few lines. I feel like because most of this is short lived (lots of feature flags removed next release), it would be ok to be very verbose and write 4 big blocks...

Let me know what you think!

Reply:
Thanks for this exhaustive list. I have added these as part of query_tab test suite.  Check descriptions below if these makes sense. These are only for `unifiedComponentsInTimelineEnabled = true`. I am trying to see if I can add tests of `unifiedComponentsInTimelineEnabled = false` but cannot promise.

![grafik](https://github.com/elastic/kibana/assets/7485038/4518b941-cebc-4fa0-a16c-7d4fa8b1fe3e)

These changes are here : 352c75aecd6

Reply:
Added for pin event as well here: https://github.com/elastic/kibana/commit/82531c17df4dd953172bbc2b30f9c73624161c32

Reply:
tested all combinations of everything and it looks perfect now, thank you!